hathor/manager.py

```diff
@@ -617,6 +620,10 @@ def _initialize_components_new(self) -> None:
         #       complex code
         self.tx_storage.indexes._manually_initialize(self.tx_storage)
 
+        # Publish events for all stored transactions
+        for tx in self.tx_storage.topological_iterator():
+            self.pubsub.publish(HathorEvents.NETWORK_NEW_TX_ACCEPTED, tx=tx)
+
```

Comment:
@jansegre, we talked about emitting those `LOAD` phase events analogously to the index reconstruction (iterating over all txs in topological order). I didn't reuse the same iteration as the index reconstruction on `IndexesManager` because it felt misplaced, since this event publishing has nothing to do with indexes. Then I replicated the iteration here, on `HathorManager` scope, but I'm not sure that is a good idea.

Also, I'm not sure if this could create any unwanted side effects, since the `NETWORK_NEW_TX_ACCEPTED` event is subscribed in multiple parts of the code.

What do you think?

Reply:
I don't feel it would have unwanted side effects, but it will have an impact on the loading time. Iterating over all transactions is a slow operation and take at least a few minutes.

Another remark: Why would we emit `NETWORK_NEW_TX_ACCEPTED` for all transactions? Shouldn't we filter out partially validated transactions? Why about voided transactions? Should we create a `LOADING_NEW_TX` event type?

Reply:
> it will have an impact on the loading time. Iterating over all transactions is a slow operation and take at least a few minutes.

I'll change it so these events are only emitted if `--emit-load-events` is enabled. (This flag already prevents _processing_ load events, but I'll change it to actually prevent this loop since it's not necessary)

> Shouldn't we filter out partially validated transactions?

Partial transactions are not iterated when using the `topological_iterator()`. This was confirmed by @jansegre and should be checked in the Sync V2 related PRs.

> Why about voided transactions?

Those are expected to be emitted, and the related metadata will contain the information that this transaction is actually voided.

>  Should we create a `LOADING_NEW_TX` event type?

It's indeed our idea to change the name of the `NETWORK_NEW_TX_ACCEPTED` event. We thought about something like `NEW_TX_ACCEPTED`, and then adding an attribute to the event body specifying its `source`, that could be for example `NETWORK`, `LOAD`, `API`, etc. This would be done in a separate, future PR.

Reply:
> I'll change it so these events are only emitted if --emit-load-events is enabled. (This flag already prevents processing load events, but I'll change it to actually prevent this loop since it's not necessary)

Done in https://github.com/HathorNetwork/hathor-core/pull/534/commits/9d2d55c6f4b961b2d39ff26197a76ea735cb5dee

Reply:
Just saw your changes on https://github.com/HathorNetwork/hathor-core/pull/534/commits/9d2d55c6f4b961b2d39ff26197a76ea735cb5dee, and it was confusing the use of it.

The way it is implemented, it seems new transactions have arrived during load time. I feel we should only publish events to the integration (and not internally into the full node). My understanding is that the use case would use this feature when they need to re-run all events for specific occasions (e.g., disaster recovery).

Sorry to bring that up now. Maybe you should remove this piece of code to get your PR merged ASAP, and discuss it further in the design or in a separate PR.

Reply:
I just finished reading all threads. This is the only point blocking me from approving the PR.

Reply:
> The way it is implemented, it seems new transactions have arrived during load time.

Even if we change the event name?

> I feel we should only publish events to the integration (and not internally into the full node).

That would be a simple change, we would just need to call the `EventManager`'s handler directly instead of publishing the event to the `PubSub`.

> Sorry to bring that up now. Maybe you should remove this piece of code to get your PR merged ASAP, and discuss it further in the design or in a separate PR.

Okay, I'll do that!

Reply:
Ok. Thanks!